README.md

```diff
@@ -1 +1,128 @@
-# grid3_client_go
\ No newline at end of file
+# **grid3_go:**
+
+grid3_go is a go client created to interact with threefold grid. It should manage CRUD operations for deployments on the grid.
+
+- ## **Grid3_go flow:**
+
+  1. the deployment manager will be initialized with identity/network information.
+  2. deployment manager will expose the `deploy` which the user will call it and give it old/new deployments
+  3. `deploy` method will do the following
+     - internally call `calculateChanges` method which
+       1. loads old deployments using their ids from teh grid
+       2. determine which things needs to be created, updated and deleted
+     - will take the sutable action for each operation to create, update and delete
+     - waits on them and report the state
+  4. For applying the changes we have subi and node pkgs which creates contracts/deployments on the grid
+  5. if a user wishes to apply any changes, they should provide their new state, and their current deployment ids.
+  6. the deployment manager should be responsible of reverting the applied changes if some error happens midway.
+
+- ### **How the deployment manager calculates changes:**
+
+- the deployment manager receives informantion about old and new deployments, then decides which operations need to be performed.
+
+  1. deployments that need to be created are present in the new deployments, and not in the old deployments.
+  2. deployments that need to be deleted are present in the old deployments, and not in the new deployments.
+  3. deployments that need to be updated are present in both old and new deployments, but they must have different hashes.
+
+- ### **Creating a new Deployment:**
+
+  1. deployments must first be signed and validated.
+  2. a deployment contract then must be created.
+  3. Then, the deployment should be deployed on the node.
+  4. If some error happens while trying to deploy on the node, the contract should be canceled to avoid leaking a contract.
```

Comment:
is it a should or a must? does the user need to do manually? does it happen internally? is it guaranteed?

Reply:
It's handled internally
It guarantees if any error happened while deploying the contract will be canceled